support-frontend/app/controllers/DigitalSubscriptionFormController.scala

```diff
@@ -0,0 +1,122 @@
+package controllers
+
+import actions.CustomActionBuilders
+import admin.settings.{AllSettings, AllSettingsProvider, SettingsSurrogateKeySyntax}
+import assets.{AssetsResolver, RefPath, StyleContent}
+import cats.implicits._
+import com.gu.identity.model.{User => IdUser}
+import com.gu.monitoring.SafeLogger
+import com.gu.monitoring.SafeLogger._
+import com.gu.support.catalog.DigitalPack
+import com.gu.support.config.{PayPalConfigProvider, StripeConfigProvider}
+import com.gu.support.pricing.PriceSummaryServiceProvider
+import com.gu.support.zuora.api.ReaderType.{Direct, Gift}
+import config.RecaptchaConfigProvider
+import controllers.UserDigitalSubscription.{redirectToExistingThankYouPage, userHasDigitalSubscription}
+import play.api.mvc._
+import play.twirl.api.Html
+import services._
+import views.EmptyDiv
+import views.html.helper.CSRF
+import views.html.subscriptionCheckout
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DigitalSubscriptionFormController(
+  priceSummaryServiceProvider: PriceSummaryServiceProvider,
+  val assets: AssetsResolver,
+  val actionRefiners: CustomActionBuilders,
+  identityService: IdentityService,
+  testUsers: TestUserService,
+  membersDataService: MembersDataService,
+  stripeConfigProvider: StripeConfigProvider,
+  payPalConfigProvider: PayPalConfigProvider,
+  components: ControllerComponents,
+  settingsProvider: AllSettingsProvider,
+  fontLoaderBundle: Either[RefPath, StyleContent],
+  recaptchaConfigProvider: RecaptchaConfigProvider
+)(implicit val ec: ExecutionContext) extends AbstractController(components) with SettingsSurrogateKeySyntax {
```

Comment:
there is still a lot of similarity between the three "formcontroller" but I haven't really touched in this PR.  Also, the main.scala.html used everywhere seems to have some required contributions specific stuff (e.g. amounts settings), but missing subs stuff (e.g. product prices and promo)  But that's not something for this PR.